Docs: Fix example path to drush.settings.php

JeremySkinner opened the pull request:
Fix missing / in example for loading drush.settings.php (`__DIR__` does not contain a trailing slash).

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/drud/ddev?pullRequest=1036) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/drud/ddev?pullRequest=1036) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/drud/ddev?pullRequest=1036) it.</sub>

JeremySkinner commented:
Thanks @rfay, I didn't realise this was auto-generated as part of settings.ddev.php, just assumed it was an incorrect example. I've gone ahead and modified the templates and removed the snippet from the docs page and replaced it with some new wording. 

I haven't changed the use of the `DEPLOY_NAME` as the way of checking we're outside the container...it seems good enough for now, and at least drush.settings.php is then usable. 

Let me know if there's anything else you'd like changed.

rfay commented:
This probably needs an automated test, but it's nontrivial to check the host-side drush usage on every platform, where we don't currently have drush installed. I *think* there are drush usage tests sprinkled around that would make sure it's working inside the web container.

JeremySkinner commented:
I'm happy to try and add some tests for drush usage on the host side, although I'm not familiar with go so might take me a bit of time.

rfay commented:
That seems like a lot to ask. When we test this we'll see if we can do some tests, if you'll allow push by rfay and andrewfrench to your fork (if you haven't already).

JeremySkinner commented:
Thanks - sounds good. Yes you should have access to push to the branch.

JeremySkinner commented on docs/users/developer-tools.md:
```diff
@@ -53,7 +53,7 @@ The port referenced is unique per running project, and randomly chosen from avai
 
 ### Using Drush Installation on Host Machine
-If you have Drush installed on your host system, you can use it to interact with a ddev project by defining a `drush.settings.php` file at the docroot of your code base, and referencing it from your `settings.php` file. The `drush.settings.php` file should look similar to below, using the host port information for your project retrieved from `ddev describe`:
+If you have Drush installed on your host system, you can use it to interact with a ddev project by defining a `drush.settings.php` file at the docroot of your code base. This will be automatically referenced from the generated `settings.ddev.php` file which is located under `web/sites/default`. The `drush.settings.php` file should look similar to below, using the host port information for your pr ...
```
I've changed to `<docroot>/sites/default`. I tested this with D8 and the include statement is automatically generated in settings.ddev.php

JeremySkinner commented on pkg/ddevapp/drupal.go:
```diff
@@ -163,7 +163,7 @@ if (empty($config_directories[CONFIG_SYNC_DIRECTORY])) {
 // it to work both within a docker container and natively on the host system.
 if (!empty($_SERVER["argv"]) && strpos($_SERVER["argv"][0], "drush") && empty($_ENV['DEPLOY_NAME'])) {
-  include __DIR__ . '../../../drush.settings.php';
+  include __DIR__ . '/../../../drush.settings.php';
```
I'd assumed there was a reason why it was referencing the doc root, rather than looking in sites/default, but if there isn't then this seems better as all the config files are then in one place.

JeremySkinner commented:
Re #941 and transient ports, personally I'd much prefer to have a single db container exposed on a single port which can then be used by all the dev sites. I know that's not the 'docker way' and others probably wouldn't like this, but would certainly fit my workflow better!

At the moment I'm just running `ddev describe` and updating the port in drush.settings.php manually whenever I start the project. Would it be feasible to automate that following a successful `ddev start`?

JeremySkinner commented:
Thanks @rfay, that looks good and will be much more useful. Looking forward to playing with it. Thanks again for getting this in so quickly 👍👍

rfay commented:
Artifacts from this PR can be downloaded for testing from https://circleci.com/gh/drud/ddev/3225#artifacts/containers/0 for testing - there are builds for macOS, Linux, and Windows there.

JeremySkinner commented:
I've been using this today and it works nicely with no issues.

rfay commented:
@andrewfrench there are probably a couple of things I slipped in here that you'll want to look at again and make sure you're good with them:

* Settings directory gitignore creation happens in app.Init() instead of in post-config, so we always write our own in there. It's the same for every CMS type.
* The settings.php or main settings file is *not* included in the gitignore. So many people actually check in a settings.php (without db credentials), that I thought it shouldn't be in there. Of course... maybe *i* added that and then reconsidered.

andrewfrench commented on pkg/ddevapp/apptypes.go:
```diff
@@ -133,6 +133,10 @@ func (app *DdevApp) CreateSettingsFile() (string, error) {
 		if err != nil {
 			util.Warning("Unable to create settings file: %v", err)
 		}
+		if err := CreateGitIgnore(filepath.Dir(app.SiteSettingsPath), filepath.Base(app.SiteLocalSettingsPath), "ddev_drush_settings.php"); err != nil {
```
This may be my CMS ignorance showing, but isn't this very drupal-centric? Do TYPO3 and Wordpress projects have a `ddev_drush_settings.php` file?

I suppose there's no harm in ignoring something that may not exist, but I think this block could be flagged for a CMS-specific refactor.

rfay replied:
My assumption is that it's ok to ignore something that doesn't exist to make the code simpler. Drush is for Drupal only, but I guess it shouldn't be offensive to people to find a non-their-CMS file gitignored.

This is actually *undoing* the previous CMS-specific approach, which involved 4x duplicated code.

andrewfrench replied:
Gotcha, I see that now. That's fine with me.

rfay commented:
This one fails with the same #fail as master on TestDdevRestoreSnapshot, but only there, and only on Windows 10 Home. We'll definitely have to find out what's going on there, hopefully tomorrow, but it's not related to this PR. I'm going to go ahead and pull it.
